packages/server/src/api/controllers/static/index.ts

```diff
@@ -126,6 +127,18 @@ export const deleteObjects = async function (ctx: Ctx) {
 }
 
 export const serveApp = async function (ctx: Ctx) {
+  // TODO: merge with middleware
+  const appDb = context.getAppDB()
+  const migrationDoc = await appDb.get<MigrationDoc>("_design/migrations")
+  const latestMigration =
+    Object.keys(MIGRATIONS).sort()[Object.keys(MIGRATIONS).length - 1]
+
+  const migrated = migrationDoc.version === latestMigration
+  if (!migrated) {
+    // TODO
+    return ctx.redirect("/app-preview-page")
```

Comment:
I don't know if we should re-direct this, as the client library won't necessarily know where to go, it might be easier to server render an app being migrated page.